### DISCUSSION TITLE
Dashboard with charts for temperature and windspeed

### DISCUSSION THREAD

**Discussion by pk1nzl** [brutella/hkknx-public#286] [General]
Hello,

anybody here who has build something like that?

Would be cool if there is a way with a script to make a dashboard showing me charts of the windspeed and temperature of the rooms and outside. Would also be nice to switch beetween the day's.

I've tried to build something with chat gpt but the go syntax compared to the built in script editor isn't really the same and i get a lot of errors.

Hope somebody can help me, maybe with an example how to make charts getting the data from group adresses.

Thank you.
Regards Philipp

**Comment by brutella** [comment]
There is currently no easy to create a full-featured dashboard using just scripts. You can create an HTTP server in a script fairly easy. But a dashboard with history charts is much more effort.

**Comment by flobxy** [comment]
Hi @pk1nzl ,
I just installed InfluxDB on my Raspberry Pi and send data from a script in hkknx to the database to visualize it. It is still work in progress but works very well.

<img width="1336" alt="Screenshot 2024-06-22 at 19 42 27" src="https://github.com/brutella/hkknx-public/assets/1721741/a875a64c-00d5-41ae-9993-66755500b941">

**Reply by pk1nzl** [reply]
Hello @flobxy
thank you very much for your feedback! (and sorry for my late answer)
Is this a script running in the hkknx gui? Can you maybe give me an example? And what is the name of the visualization you used?

**Reply by flobxy** [reply]
Hi,
Happy to upload the script. It’s super straight forward. Listening to the knx telegrams and if energy arrives, convert and push to influx. I am then visualizing the data in influx on its own server running on same raspi. 

Please note with the latest update you can visualize any data points within hkknx gui. Just search for the device and check the graphs.

**Reply by flobxy** [reply]
```go
var influxdb, hkknx = import("influxdb"), import("hkknx")
var fmt, time = import("fmt"), import("time")
var math = import ("math")

// Create an influxdb client
var client = influxdb.NewClient("http://192.168.178.250:8086", "your secret key")
var bucket = client.Bucket("FloRaspi64", "knxv3")

for {
	wind = hkknx.GroupWriteNotify("0/1/7")
	temp = hkknx.GroupWriteNotify("0/1/8")
	rain = hkknx.GroupWriteNotify("0/1/9")
	light = hkknx.GroupWriteNotify("0/1/14")
	vebrauch_kw = hkknx.GroupWriteNotify("0/1/21")
	vebrauch_kwh = hkknx.GroupWriteNotify("0/1/23")
	ueberschuss_kw = hkknx.GroupWriteNotify("0/1/22")
	ueberschuss_kwh = hkknx.GroupWriteNotify("0/1/24")
	
for { // Wait for any of the events to happen.
    select {
	case <-vebrauch_kw:
    		var msg, err = hkknx.GroupRead("0/1/21")
    		var buf = hkknx.ParseDPT14056(msg) //convert to float
    		var point = influxdb.NewPointWithMeasurement("energy")
    		point.AddTag("address", "Vebrauch_in_kw")
    		point.AddField("value", buf)
			bucket.Write(point)
	case <-vebrauch_kwh:
    		var msg, err = hkknx.GroupRead("0/1/23")
    		var buf = hkknx.ParseDPT13010(msg) //convert to float
    		var point = influxdb.NewPointWithMeasurement("energy")
    		point.AddTag("address", "Vebrauch_in_kwh")
    		point.AddField("value", math.Abs(buf)/1000)
    		bucket.Write(point)
	case <-ueberschuss_kw:
    		var msg, err = hkknx.GroupRead("0/1/22")
    		var buf = hkknx.ParseDPT14056(msg) //convert to float
    		var point = influxdb.NewPointWithMeasurement("energy")
    		point.AddTag("address", "Uberschuss_in_kw")
    		point.AddField("value", buf)
    		bucket.Write(point)
	case <-ueberschuss_kwh:
    		var msg, err = hkknx.GroupRead("0/1/24")
    		var buf = hkknx.ParseDPT13010(msg) //convert to float
    		var point = influxdb.NewPointWithMeasurement("energy")
    		point.AddTag("address", "Ueberschuss_in_kwh")
    		point.AddField("value", math.Abs(buf)/1000)
    		bucket.Write(point)
	case <-wind:
    		var msg, err = hkknx.GroupRead("0/1/7")
    		var buf = hkknx.ParseDPT9005(msg) //convert to float
		   // Convert meter per second to kilometer per hour.
        	windSpeed = buf * 3.6
    		var point = influxdb.NewPointWithMeasurement("windspeed")
    		point.AddTag("address", "meteodata")
    		point.AddField("value", windSpeed)
			bucket.Write(point)
	case <-temp:
    		var msg, err = hkknx.GroupRead("0/1/8")
    		var buf = hkknx.ParseDPT9001(msg) //convert to float
    		var point = influxdb.NewPointWithMeasurement("temperatur")
    		point.AddTag("address", "meteodata")
    		point.AddField("value", buf)
    		bucket.Write(point)
	case <-light: //buf = <-temp:
    		var msg, err = hkknx.GroupRead("0/1/14")
    		var buf = hkknx.ParseDPT9004(msg) //convert to float
    		var point = influxdb.NewPointWithMeasurement("lux")
    		point.AddTag("address", "meteodata")
    		point.AddField("value", buf)
    		bucket.Write(point)
        }
    }
}
```

**Reply by pk1nzl** [reply]
Thank you very much @flobxy, i will try this as soon as possible...!

**Reply by flobxy** [reply]
Also try out the new built in functionality from Brutella (in next official release or beta). It might be sufficient for your needs.

**Reply by pk1nzl** [reply]
Yes, the new release is almost that what i need. But for example Windspeed is one thing i am missing currently, also the energy data as you have in your script would be nice.
Anyway, i will try your script.

@brutella do you have windspeed on the roadmap for monitoring, or what other data do you have planned to show in monitoring?
Would be interesting.
Thx

**Reply by brutella** [reply]
In version 3, if you have a group address with DPT 9.005 (Windspeed), you will see the wind speeds in a chart as shown in https://github.com/brutella/hkknx-public/discussions/314#discussioncomment-9882096

You don't have to explicitly create an accessory because HomeKit doesn't "understand" the characteristic *wind speed* anyway. You only have to make sure that you define the correct datapoint type in your ETS project.

**Comment by brutella** [comment]
In version 3, if you have a group address with DPT 9.005 (Windspeed), you will see the wind speeds in a chart as shown in https://github.com/brutella/hkknx-public/discussions/314#discussioncomment-9882096

You don't have to explicitly create an accessory because HomeKit doesn't "understand" the characteristic *wind speed* anyway. You only have to make sure that you define the correct datapoint type in your ETS project.
